src/components/SiteDetail.js

```diff
@@ -43,14 +41,13 @@ class SiteDetail extends Component {
     this.loadSiteData();
   }
 
-  async loadSiteData() {
+  loadSiteData() {
     const { selectSite } = this.props;
     const { loadedSite } = this.state;
 
     if (selectSite) {
       if (!loadedSite || (loadedSite && loadedSite.id !== selectSite.id)) {
```

Comment:
A thought here on this old code:

 `(!A or [ A and A.element differs from B.element ])` is equivalent to `(!A or [A.element differs from B.element])`